crates/chain/src/indexed_tx_graph.rs

```diff
@@ -145,27 +141,149 @@ where
         // 2. decide whether to insert them into the graph depending on whether `is_tx_relevant`
         //    returns true or not. (in a second loop).
         let mut changeset = ChangeSet::<A, I::ChangeSet>::default();
-        let mut transactions = Vec::new();
-        for (tx, anchors) in txs.into_iter() {
+
+        let txs = txs
+            .into_iter()
+            .inspect(|(tx, _, _)| changeset.indexer.append(self.index.index_tx(tx)))
+            .collect::<Vec<_>>();
+
+        for (tx, anchors, seen_at) in txs {
+            if self.index.is_tx_relevant(tx) {
+                changeset.append(self.insert_tx(tx, anchors, seen_at));
+            }
+        }
+
+        changeset
+    }
+
+    /// Batch insert transactions.
+    ///
+    /// All transactions in `txs` will be inserted. To filter out irrelevant transactions, use
+    /// [`batch_insert_relevant`] instead.
+    ///
+    /// [`batch_insert_relevant`]: IndexedTxGraph::batch_insert_relevant
+    pub fn batch_insert<'t>(
```

Comment:
Don't think this method is worth it. Doesn't do anything beyond calling `insert_tx` in a loop and appending changesets (it's surprising that it's not implemented like this).

Reply:
Do you mind expanding on what you mean by what's not implemented like this?

Reply:
I think I understand what you mean now. This method is now removed.